### PR TITLE
MRKT-78: similar artist section to API data

### PR DESCRIPTION
**Pull request by scandycuz**

### CODE REVIEW

File: apps/marketplace/src/components/SimilarArtists.tsx [hunk 1]
```diff
@@ -1,54 +1,29 @@
...
-              );
-            }
-          ) }
-      </Grid>
+      <Artists
+        artists={ artists }
+        hasTitle={ true }
+        isLoading={ isLoading }
+        itemOrientation="row"
+        numSkeletons={ 6 }
```
Comment (escobarjonatan) [inline apps/marketplace/src/components/SimilarArtists.tsx]: We could have this 6 be a variable and we use it here and in the limit as well.

File: apps/marketplace/src/components/SimilarArtists.tsx [hunk 2]
```diff
@@ -1,54 +1,29 @@
...
-    setTimeout(() => {
-      setIsLoading(false);
-    }, 1000);
-  }, []);
+const SimilarArtists: FunctionComponent<SimilarArtistsProps> = ({ genre }) => {
+  // TODO: limit to artists with marketplace sales when API updated
+  const { isLoading, data: artists = [] } = useGetArtistsQuery({
+    genres: genre ? [genre] : [],
+    limit: 6,
+  });
```
Comment (escobarjonatan) [inline apps/marketplace/src/components/SimilarArtists.tsx]: Do we need to add anything to exclude the current artist or will it automatically do that?

File: apps/marketplace/src/components/Artists.tsx [hunk 3]
```diff
@@ -49,29 +50,37 @@ const Artists: FunctionComponent<ArtistsProps> = ({
...
+            return (
+              <Grid
+                display="flex"
+                justifyContent="center"
+                key={ id }
+                lg={ gridSizeColumnMap.lg[itemOrientation] }
+                md={ gridSizeColumnMap.md[itemOrientation] }
+                sm={ gridSizeColumnMap.sm[itemOrientation] }
+                xs={ gridSizeColumnMap.xs[itemOrientation] }
+                item
```
Comment (escobarjonatan) [inline apps/marketplace/src/components/Artists.tsx]: Is this a Boolean? If so, we can make it explicit item={true}

**Review by escobarjonatan — APPROVED** [review]

File: apps/marketplace/src/components/Artists.tsx (hunk 3 above)
Comment (scandycuz) [reply apps/marketplace/src/components/Artists.tsx]: I'm ok with that. We should probably update the eslint config if we want to standardize on that.

File: apps/marketplace/src/components/SimilarArtists.tsx (hunk 2 above)
Comment (scandycuz) [reply apps/marketplace/src/components/SimilarArtists.tsx]: Good point. It should be returning the current artist without that. I'll create a bug ticket.

File: apps/marketplace/src/components/MoreSongs.tsx [hunk 4]
```diff
@@ -37,7 +42,7 @@ const MoreSongs: FunctionComponent<MoreSongsProps> = ({
 
   return (
     <Box mt={ [10, 8, 12.5] }>
-      <Sales isLoading={ isLoading } sales={ sales } title={ title } />
+      <Sales isLoading={ skip || isLoading } sales={ sales } title={ title } />
```
Comment (escobarjonatan) [inline apps/marketplace/src/components/MoreSongs.tsx:45]: if `skip` is true, it means no data will be fetch and will be loading until user refreshes? might be worth handling that separate and show some text to action the user to refresh?

Comment (scandycuz) [reply apps/marketplace/src/components/MoreSongs.tsx:45]: Without `isLoading={ skip || isLoading }`, the UI will show a "no results" state and then a loading state. This ensures that the loading state is present even when the params to fetch the data aren't present yet, and that no uncecessary requests are made before the params are present.

Comment (scandycuz) [reply apps/marketplace/src/components/MoreSongs.tsx:45]: If there's some sort of error and the request is never actually made, the user should see a toast that a request failed. We could look into adding some some sort of error state in the UI in the future for that use case.

File: apps/marketplace/src/components/SimilarArtists.tsx (hunk 2 above)
Comment (scandycuz) [reply apps/marketplace/src/components/SimilarArtists.tsx]: The bug's been resolved.

**Pull request merged**